### PR TITLE
[Helix] dmp/crash tweaks

### PR DESCRIPTION
**Pull request by HaoK**
- Specify dmp output location directly and get rid of copy
- Generate diagnostic logs which can help in some hangs
- Upload sequence file which helps with some hangs as well

### CODE REVIEW

File: eng/helix/content/RunTests/TestRunner.cs [hunk 1]
```diff
@@ -322,21 +323,19 @@ public void UploadResults()
             {
                 Console.WriteLine("No logs found in artifacts/log");
             }
-            Console.WriteLine($"Copying TestResults/**/*.dmp to {HELIX_WORKITEM_UPLOAD_ROOT}/");
+            Console.WriteLine($"Copying TestResults/**/Sequence*.xml to {HELIX_WORKITEM_UPLOAD_ROOT}/");
```
Comment (dougbu) [inline eng/helix/content/RunTests/TestRunner.cs:326]: So, what are Sequence*.xml files❔

File: eng/helix/content/RunTests/TestRunner.cs [hunk 2]
```diff
@@ -322,21 +323,19 @@ public void UploadResults()
...
                     Console.WriteLine($"Copying: {file} to {Path.Combine(HELIX_WORKITEM_UPLOAD_ROOT, fileName)}");
-                    // Need to copy to HELIX_WORKITEM_UPLOAD_ROOT and HELIX_WORKITEM_UPLOAD_ROOT/../ in order for Azure Devops attachments to link properly and for Helix to store the logs
                     File.Copy(file, Path.Combine(HELIX_WORKITEM_UPLOAD_ROOT, fileName));
-                    File.Copy(file, Path.Combine(HELIX_WORKITEM_UPLOAD_ROOT, "..", fileName));
                 }
             }
             else
             {
-                Console.WriteLine("No dmps found in TestResults");
+                Console.WriteLine("No sequence xml file found in TestResults");
```
Comment (dougbu) [inline eng/helix/content/RunTests/TestRunner.cs:338]: This message involves incorrect assumptions. The entire TestResults/ folder is missing.

File: eng/helix/content/runtests.cmd [hunk 3]
```diff
@@ -20,6 +20,7 @@ set DOTNET_ROOT=%DOTNET_HOME%\%$arch%
 set DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
 set DOTNET_MULTILEVEL_LOOKUP=0
 set DOTNET_CLI_HOME=%CD%\home%random%
+set VSTEST_DUMP_PATH=%HELIX_DUMP_FOLDER%
```
Comment (dougbu) [inline eng/helix/content/runtests.cmd:23]: To confirm: `dotnet test` uses this environment variable for created dumps automatically❔

Comment (HaoK) [reply eng/helix/content/runtests.cmd:23]: Yeah they added support to control where they directly output dumps, so this makes it so we don't have to move them to the helix upload root manually

File: eng/helix/content/RunTests/TestRunner.cs (hunk 1 above)
Comment (HaoK) [reply eng/helix/content/RunTests/TestRunner.cs:326]: You can look at them to see what tests ran and which didn't finish, its another way to look for hanging tests in the cases where there its not obvious from the logs.  There are some cases where sequence.xml doesn't get written out either, and in those cases you have to go all the way down to the detailed diagnostic logs: vstest.log

Comment (HaoK) [reply eng/helix/content/RunTests/TestRunner.cs:326]: We shouldn't have to go digging into either of these files normally, but it gives us the most diagnostics we can get right now for crashes/hangs

**Comment by HaoK** [comment]
@dougbu forgot about these dmp tweaks for ages, can you take a final look?

**Review by dougbu — APPROVED** [review]
looks fine to me. Did you test collection w/ a forced dump or two❔

**Comment by HaoK** [comment]
Yeah I did back in the 2020 when I first opened these, this was the last iteration of the dmp/crash work, switching to the new variables/etc.

**Comment by HaoK** [comment]
Also, this is how we get most of our cores today anyways, since we don't copy from the HELIX_DUMP_FOLDER to the uploads

**Pull request merged**